docs/DeveloperGuide.md

```diff
@@ -158,47 +159,40 @@ Classes used by multiple components are in the `seedu.address.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Undo feature
 
-#### Proposed Implementation
+#### Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The undo mechanism is facilitated by the abstract class `ConcreteCommand`. 
+It extends `Command` and has the `undo()` method. The `undo()` method is called when the user executes the `undo` command. 
+The `undo()` method reverses the effects of the command that was previously executed.
+The `undo()` method is implemented in the concrete command classes, such as `AddCommand`, `DeleteCommand`, `EditCommand`, etc.
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+The `Model` component stores a history of executed concrete commands in a stack.
+When a command is executed successfully, the command is pushed onto the stack.
+When the user executes the `undo` command, the `Model` component pops the last command from the stack and calls the `undo()` method of the command.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+These operations are exposed in the `Model` interface as `Model#pushToUndoStack()`, `Model#undoAddressBook()`.
```

Comment:
Would it be better if its `Model#pushToUndoStack()` and `Model#undoAddressBook()`?

Reply:
It is `Model#pushToUndoStack()` and `Model#undoAddressBook()`.